File: VB/WindowsApplication3/ToggleSwitch/CheckEditViewInfoDescendant.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.Utils.Drawing

Namespace WindowsApplication3
	Public Class ToggleSwitchViewInfo
		Inherits CheckEditViewInfo
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub

		Protected Overrides Sub UpdateCheckProperties(ByVal e As DevExpress.Utils.Drawing.BaseCheckObjectInfoArgs)
			MyBase.UpdateCheckProperties(e)
			Dim args As CheckObjectInfoArgs = TryCast(e, CheckObjectInfoArgs)
			If args.CheckState = System.Windows.Forms.CheckState.Checked Then
				e.Caption = Item.OnText
			Else
				e.Caption = Item.OffText
			End If
		End Sub

		Private Shadows ReadOnly Property Item() As RepositoryItemToggleSwitch
			Get
				Return TryCast(MyBase.Item, RepositoryItemToggleSwitch)
			End Get
		End Property
		Public Overrides ReadOnly Property IsSupportFastViewInfo() As Boolean
			Get
				Return False
			End Get
		End Property
	End Class
End Namespace

File: CS/WindowsApplication3/ToggleSwitch/CheckEditViewInfoDescendant.cs
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.Utils.Drawing;

namespace WindowsApplication3 {
    public class ToggleSwitchViewInfo : CheckEditViewInfo {
        public ToggleSwitchViewInfo(RepositoryItem item)
            : base(item) {
        }
      
        protected override void UpdateCheckProperties(DevExpress.Utils.Drawing.BaseCheckObjectInfoArgs e)
        {
            base.UpdateCheckProperties(e);
            CheckObjectInfoArgs args = e as CheckObjectInfoArgs;
            if (args.CheckState == System.Windows.Forms.CheckState.Checked)
                e.Caption = Item.OnText;
            else
                e.Caption = Item.OffText;
        }

        new RepositoryItemToggleSwitch Item {
            get { return base.Item as RepositoryItemToggleSwitch; }
        }
        public override bool IsSupportFastViewInfo {
            get { return false; }
        }
    }
}
